### DISCUSSION TITLE
Bug: new drawing button broken for long sheet and clear button not working with selections

### DISCUSSION THREAD

**Discussion by Lioncub901** [keiwando/toonsquid#81] [Q&A] [answered]
When I change the setting to a drawing for 2 frame(animating in twos) and add a drawing it will not add a new 2 length drawing after the 2 length current drawing but a 1 length drawing in the 2 length current drawing
 (Split the current drawing).

Also when I duplicate a 2 length frame it does not duplicate the frame after the current frame but in the frame and the duplicated frame is a 1 length frame

Also when I select something and click clear in the drawing layer panel it will clear the whole layer and not the selected area.

**Comment by Lioncub901** [comment]
Have any of these issues been resolved?

**Comment by keiwando** [comment, marked as answer]
Hi! 
None of these are technically bugs since all of the behaviour is working as currently intended. 
The clearing of the selection area using the "Clear" action on the layer is a feature request that is already planned and I can definitely also see the value of the "add drawing" behaviour. To be clear, would a setting be enough with which you could say that the "Add Drawing" button should never split an existing drawing but always add it on the current frame if there is no drawing there yet and otherwise after the existing drawing at the default length that you chose in the settings? Or do you have cases where you want it to still split the current drawing?

**Comment by Lioncub901** [comment]
@keiwando I think having the new drawing inserted after the current one with the same length is better because it will allow people to add in betweens.
